mkosi

```diff
@@ -1027,8 +1027,14 @@ def reset_machine_id(args, workspace, run_build_script):
         os.unlink(machine_id)
         open(machine_id, "w+b").close()
         dbus_machine_id = os.path.join(workspace, 'root', 'var/lib/dbus/machine-id')
-        if os.path.exists(dbus_machine_id):
+        has_dbus_machine_id = False
+        try:
             os.unlink(dbus_machine_id)
+            has_dbus_machine_id = True
+        except FileNotFoundError:
+            pass
+
+        if has_dbus_machine_id:
```

Comment:
i figure that was my bad suggestion. I forgot that you can do the "else" syntax when catching exceptions...